developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -42,50 +42,10 @@ const baseOptions: CompilerOptions = {
  */
 let callbackProcIdentifier = 0;
 
-/**
- * Pointer in wasm-space
- */
-type WasmPtr = number;
-
-/**
- * The wrapped functions
- */
-class WasmWrapper {
-  Module: any;
-
-  compileKeyboardFile?: (pszInfile: string, pszOutfile: string, aSaveDebug: number, aCompilerWarningsAsErrors: number, aWarnDeprecatedCode: number, msgProc: string) => boolean;
-  parseUnicodeSet?: (pat: string, buf: WasmPtr, length: number) => number;
-  setCompilerOptions?: (shouldAddCompilerVersion: number) => boolean;
-
-  constructor(wasmModule: any) {
-    this.Module = wasmModule;
-    if (!wasmModule) {
-      throw Error(`wasm host did not load`);
-    }
-    this.compileKeyboardFile = this.Module.cwrap('kmcmp_Wasm_CompileKeyboardFile', 'boolean', ['string', 'string', 'number', 'number', 'number', 'string']);
-    this.parseUnicodeSet = this.Module.cwrap('kmcmp_Wasm_ParseUnicodeSet', 'number', ['string', 'number', 'number']);
-    this.setCompilerOptions = this.Module.cwrap('kmcmp_Wasm_SetCompilerOptions', 'boolean', ['number']);
-
-    if (this.parseUnicodeSet === undefined
-      || this.setCompilerOptions === undefined
-      || this.compileKeyboardFile === undefined) {
-        throw Error(`some wasm functions did not load properly.`);
-    }
-  }
-
-  /**
-   * Entry point into Wasm functions
-   * @returns WasmWrapper
-   */
-  public static async load() : Promise<WasmWrapper> {
-    return new WasmWrapper(await loadWasmHost());
-  }
-};
-
 export class KmnCompiler {
+  private Module: any;
```

Comment:
We can't define an interface type of the expected format and type-cast to that?  Sure, maybe we can't define a _class_, but _interface_ use should be 100% fine.  Can even declare expected methods.

Reply:
Definitely not suggested for this PR, but a thread that may or may not have useful information for automating .d.ts for our WASM compiles:  https://github.com/emscripten-core/emscripten/issues/7083

Reply:
Hand-crafting an interface here is of limited value -- it still would have to be manually compared to the C++ code anyway.

Runtime checks of the interface (as exercised by unit tests) already give us confidence that the interfaces are correct.

Note that the module is by design _only_ referenced in this one class, so verification of interface changes is kept as minimal as possible.

Yes, there is some local benefit in automating the types for the interface between WASM and TS but the interface is simple enough at this time to leave it for now.